File: ch2/lineards/UVa10925.java
import java.io.*;
import java.util.*;
import java.math.BigInteger;                     // in package java.math

class Main {                                     // UVa 10925 - Krakovia
  public static void main(String[] args) throws Exception {
    BufferedReader br = new BufferedReader(      // use BufferedReader
      new InputStreamReader(System.in));
    PrintWriter pw = new PrintWriter(            // and PrintWriter
      new BufferedWriter(new OutputStreamWriter(System.out))); // = fast IO
    int caseNo = 0;
    while (true) {
      StringTokenizer st = new StringTokenizer(br.readLine());
      int N = Integer.parseInt(st.nextToken());  // N bills
      int F = Integer.parseInt(st.nextToken());  // F friends
      if (N == 0 && F == 0) break;
      BigInteger sum = BigInteger.ZERO;          // built-in constant
      for (int i = 0; i < N; ++i) {              // sum the N large bills
        BigInteger V = new BigInteger(br.readLine()); // string constructor
        sum = sum.add(V);                        // BigInteger addition
      }
      pw.printf("Bill #%d costs ", ++caseNo);
      pw.printf(sum.toString());
      pw.printf(": each friend should pay ");
      pw.printf(sum.divide(BigInteger.valueOf(F)).toString());
      pw.printf("\n\n");                         // divide to F friends
    }
    pw.close();
  }
}

File: ch2/lineards/UVa10925.py
import sys
inputs = sys.stdin.read().splitlines()           # make Python I/O faster
caseNo = 1
ln = 0
while True:
    N, F = map(int, inputs[ln].split())          # N bills, F friends
    ln += 1
    if N == 0 and F == 0: break
    sum = 0                                      # native support
    for _ in range(N):                           # sum the N large bills
        sum += int(inputs[ln])                   # native Big Integer
        ln += 1
    print("Bill #%d costs %d: each friend should pay %d\n"
        % (caseNo, sum, sum//F))                 # integer division
    caseNo += 1
